atlasdb-buggify/src/main/java/com/palantir/atlasdb/buggify/api/BuggifyFactory.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * (c) Copyright 2023 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.buggify.api;
+
+public interface BuggifyFactory {
+    /**
+     * A Buggify instance that may or may not perform any actions.
+     */
+    Buggify maybe(double probability);
```

Comment:
We could use `Optional` here instead, but then kind of forces the behavior of all-or-nothing. In the future, we could probably get a bit more creative here, and may return an instance that actually executes the buggify runnable twice, or something to that effect. Plus, it's a bit cleaner I find, but realize this is subjective, so defer to the reviewer -- don't hold strong objections to making this `Optional`.